src/action/payment.js

```diff
@@ -64,15 +64,23 @@ class PaymentAction {
 
   async decodeInvoice({ invoice }) {
     try {
+      const timer = setTimeout(() => this._nav.goWait(), WAIT_DELAY);
       const { payment, settings } = this._store;
       const request = await this._grpc.sendCommand('decodePayReq', {
         pay_req: invoice.replace(PREFIX_URI, ''),
       });
+      const fee = await this.estimateLightningFee({
+        destination: request.destination,
+        satAmt: request.num_satoshis,
+      });
       payment.amount = toAmount(parseSat(request.num_satoshis), settings.unit);
       payment.note = request.description;
+      payment.fee = toAmount(parseSat(fee), settings.unit);
```

Comment:
This function just decodes the invoice and returns a boolean.

I'd set `store.payment.fee` in `estimateLightningFee` itself and call it after `this._nav.goPayLightningConfirm()` here https://github.com/lightninglabs/lightning-app/pull/448/files#diff-c103796afd01e866f6af483b8930cf32R57.